methods/enumerables.rb

```diff
@@ -0,0 +1,98 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    size.times do |index|
+      yield self[index]
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each) unless block_given?
+
+    size.times do |index|
+      yield self[index], index
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    answer = []
+    my_each do |item|
+      answer.push(item) if yield item
+    end
+    answer
+  end
+
+  def my_all?
+    return true unless block_given?
+
+    my_each do |item|
+      return false unless yield item
+    end
+    true
+  end
+
+  def my_any?
+    return true unless block_given?
+
+    my_each do |item|
+      return true unless yield item
+    end
+    false
+  end
```

Comment:
## my_any?:

- returns true if the block ever returns a value other than false or nil.

```ruby

     array = [3,4 ,43,2,1]
     true_block = proc { |num| num <= 55 }
     array.my_any?(&true_block)).to eq(array.any?(&true_block)

```

- returns false if the block ever returns a value false or nil:

```ruby

     array = [3,4 ,43,2,1]
     false_block = proc { |num| num > 55 }
     array.my_any?(&false_block) === array.any?(&false_block) => false

```

- you need to support ranges also here.

- when no block or argument is given returns false if at least one of the collection is not true:

```ruby

     false_array = [nil, false, 1, 2, 3]
     false_array.my_any? === false_array.any? => false

```

- when a class is passed as an argument returns true if at least one of the collection is a member of such class::Integer.


- when a Regex is passed as an argument returns true if any of the collection matches the Regex.

- when a Regex is passed as an argument returns false if none of the collection matches the Regex.

- when a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern:

```ruby

     words = %w[dog door rod blade]
     words.my_any?('cat') === words.any?('cat') => false

```